util/redirectUser.js

```diff
@@ -0,0 +1,17 @@
+/*
+ * Function to redirect user, within getServerSideProps functions.
+ * Leaving it blank will error out
+ * @param  {string}    arg1 The intended link to access. examples: '/error', '/admin', '/class', '/'.
+ * @return {[type]}    HTTP request to redirected link
+ */
+
+const redirectUser = destination => {
+  return {
+    redirect: {
+      destination: destination,
+      permanent: false
+    }
+  };
+};
+
+export default redirectUser;
```

Comment:
Tested the PR by doing the following (all checks resulted as expected):

1. Initially signing in as user ‘NONE’ which is the default user when first signing on to the app -> this correctly showed the ‘Access Denied’ UI as intended
<img width="1032" alt="Screenshot 2024-02-10 at 6 33 54 PM" src="https://github.com/freeCodeCamp/classroom/assets/20196901/1628946e-9e0a-4954-b177-df63ce6b1b35">

2. Initially signing in as user ‘STUDENT’ -> this correctly showed the ‘Access Denied’ UI as intended
<img width="1032" alt="Screenshot 2024-02-10 at 6 33 54 PM" src="https://github.com/freeCodeCamp/classroom/assets/20196901/416b7696-a929-4d4d-9686-059b5619490a">



3. Initially signing in as user ‘TEACHER’ -> this correctly shows the class creation UI as intended
<img width="1012" alt="Screenshot 2024-02-10 at 6 33 02 PM" src="https://github.com/freeCodeCamp/classroom/assets/20196901/b501a9ec-b776-4699-9eb9-8388c8d8455d">

4. Initially signing in as user ‘ADMIN’ -> this correctly displays the admin dashboard
<img width="1029" alt="Screenshot 2024-02-10 at 6 38 55 PM" src="https://github.com/freeCodeCamp/classroom/assets/20196901/05d89a82-2691-45b0-95ee-fe0e94f4f083">

Edge case to consider:
When updating the user privileges in Prisma, the UI does not update immediately as the link is still set to [https://{CODESPACES}.app.github.dev/error](about:blank). This could mean that logic may need to be added which would update the URL (‘error’, ‘admin’, ‘teacher’, ‘student’) once a user privilege is updated.

Mentioning @utsab for further view.